tpm2/test/tpm/object_tools_test.c

```diff
@@ -0,0 +1,236 @@
+//############################################################################
+// object_tools_test.c
+//
+// Tests for TPM 2.0 object utility functions in tpm2/src/tpm/object_tools.c
+//############################################################################
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <CUnit/CUnit.h>
+
+#include "defines.h"
+#include "tpm2_interface.h"
+
+#include "object_tools_test.h"
+#include "object_tools.h"
+
+//----------------------------------------------------------------------------
+// object_tools_add_tests()
+//----------------------------------------------------------------------------
+int object_tools_add_tests(CU_pSuite suite)
+{
+  if (NULL == CU_add_test(suite, "init_kmyth_object_sensitive() Tests",
+                          test_init_kmyth_object_sensitive))
+  {
+    return 1;
+  }
+  if (NULL == CU_add_test(suite, "init_kmyth_object_template() Tests",
+                          test_init_kmyth_object_template))
+  {
+    return 1;
+  }
+  if (NULL == CU_add_test(suite, "init_kmyth_objec_attributes() Tests",
+                          test_init_kmyth_object_attributes))
+  {
+    return 1;
+  }
+  if (NULL == CU_add_test(suite, "init_kmyth_object_parameters() Tests",
+                          test_init_kmyth_object_parameters))
+  {
+    return 1;
+  }
+  if (NULL == CU_add_test(suite, "init_kmyth_object_unique() Tests",
+                          test_init_kmyth_object_unique))
+  {
+    return 1;
+  }
+
+  return 0;
+}
+
+//----------------------------------------------------------------------------
+// test_init_kmyth_object_sensitive
+//----------------------------------------------------------------------------
+void test_init_kmyth_object_sensitive(void)
+{
+  TPM2B_AUTH object_auth = {0};
+  TPM2B_SENSITIVE_CREATE sensitiveArea = {0};
+
+  // A null sensitive area should produce an error
+  CU_ASSERT(init_kmyth_object_sensitive(object_auth, (uint8_t *) NULL, 0,
+                                        (TPM2B_SENSITIVE_CREATE *) NULL) == 1);
+
+  // Empty object auth and data should yield a valid sensitive area
+  CU_ASSERT(init_kmyth_object_sensitive(object_auth, (uint8_t *) NULL, 0,
+                                        &sensitiveArea) == 0);
+  CU_ASSERT(sensitiveArea.sensitive.userAuth.size == 0);
+  CU_ASSERT(sensitiveArea.sensitive.data.size == 0);
+  CU_ASSERT(sensitiveArea.size == 4);
+
+  // Non-empty object auth and data should yield a valid sensitive area
+  object_auth.size = 4;
+  uint8_t object_data[] = {1, 2, 3, 4};
+  size_t object_dataSize = 4;
+
+  CU_ASSERT(init_kmyth_object_sensitive(object_auth, object_data,
+                                        object_dataSize, &sensitiveArea) == 0);
+  CU_ASSERT(sensitiveArea.sensitive.userAuth.size == 4);
```

Comment:
This is kind of a trivial comment, but since we setup a parameterized test case here, should we use the variables in the assertions? That way, we could potentially alter the test case definition without affecting the test itself. In my subjective opinion, using the variable names also makes it slightly more understandable:
- CU_ASSERT(sensitiveArea.sensitive.userAuth.size == object_auth.size)
- CU_ASSERT(sensitiveArea.data.size == object_dataSize)
- CU_ASSERT(sensitiveArea.size = object_auth.size + object_dataSize + 4)

Reply:
Done